table/block_based_table_reader.h

```diff
@@ -495,6 +496,11 @@ struct BlockBasedTable::Rep {
   // A value of kDisableGlobalSequenceNumber means that this feature is disabled
   // and every key have it's own seqno.
   SequenceNumber global_seqno;
+
+  // If false, blocks in this file are definitely all uncompressed. Knowing this
+  // before reading individual blocks enables certain optimizations.
+  bool blocks_maybe_compressed = true;
```

Comment:
Rather than having it as an option, I suggest we make it a table property and read it when opening a specific SST file.

Reply:
Ignore what I said. I misread your code.